cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -327,6 +333,10 @@ private void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Sec
 
     Future<Void> clusterOperatorSecret(Clock clock) {
         oldCoSecret = clusterCa.clusterOperatorSecret();
+        if (!this.isClusterCaFullyTrusted) {
+            LOGGER.warnCr(reconciliation, "Cluster CA not fully trusted, keeping current CO secret and certs");
+            return Future.succeededFuture();
+        }
```

Comment:
Should there be some check for the secret's existence? I.e. what happens if the secret is deleted during a renewal (if the CA is already expired, that is a common approach to delete all secrets and have the operator recreate them).

Reply:
Yeah, I added it.